llvm/unittests/Support/MustacheTest.cpp

```diff
@@ -0,0 +1,984 @@
+//===- llvm/unittest/Support/MustacheTest.cpp ----------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Test conforming to Mustache 1.4.2 spec found here:
+// https://github.com/mustache/spec
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm/Support/Mustache.h"
+#include "gtest/gtest.h"
+
+using namespace llvm;
+using namespace llvm::mustache;
+using namespace llvm::json;
+
```

Comment:
Overall I think the tests look good. I think we should try and make it obvious which parts of the spec are being tested, either through comments or test names. Comments listing the part of the spec under test seem fine, since I imaging multiple test fixtures would fall under the same category. Up to you though.

You may want to consider GTEST Fixtures for test organization, and may also want to consider how similar many of the tests are (e.g. only differ in input and expected output but otherwise have the same body), and implement some helper functions or other GTEST features to make them easier to update/maintain.